src/macos/View.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+* View.tsx
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* Mac-specific implementation of View.
+*/
+import React = require('react');
+import Types = require('../common/Types');
+
+import { View as ViewCommon } from '../native-common/View';
+
+export class View extends ViewCommon {
+
+    protected _buildInternalProps(props: Types.ViewProps) {
+        // Base class does the bulk of _internalprops creation
+        super._buildInternalProps(props);
+
+        // Drag and drop related properties
+        for (const name of ['onDragEnter', 'onDragOver', 'onDrop', 'onDragLeave']) {
+            const handler = this._internalProps[name];
+
+            if (handler) {
+                this._internalProps.allowDrop = true;
+
+                this._internalProps[name] = (e: React.SyntheticEvent<View>) => {
+                    handler({
+                        dataTransfer: (e.nativeEvent as any).dataTransfer,
```

Comment:
You are not filling in any of the other fields in the SyntheticEvent. DragEvent derives from MouseEvent, which derives from SyntheticEvent. Each of these adds a bunch of fields including mouse positions, modifier key states, etc.

Reply:
I used EventHelpers.toMouseEvent to copy missing fields, I can leave setting dataTransfer fields here or just move it all to a new EventHelpers method, since the same issue is in the Windows implementation. What do you prefer?

Reply:
Might as well create a new EventHelpers method that copies the dataTransfer after it calls toMouseEvent. That way we minimize code duplication.

Reply:
Done.